### PR TITLE
Bug 1827585: Add controller that watches leader changes and capture disk metrics

### PR DESCRIPTION
**Pull request by mfojtik**
The controller added here will watch the `etcd_server_leader_changes_seen_total` metrics and get the value for last 5 minutes. If leader changes are seen (value>0), then it will query prometheus to get `etcd_disk_wal_fsync_duration_seconds_bucket` for last 5 minutes.

This is then reported as Warning event for etcd operator. In future this can become a degraded condition.

### CODE REVIEW

**Comment by mfojtik** [comment]
@deads2k @hexfusion lets finish this :-)

**Comment by mfojtik** [comment]
only last commit has the change

**Comment by mfojtik** [comment]
@stevekuznetsov any clues about what is happening here?

**Comment by mfojtik** [comment]
@hexfusion @deads2k @alaypatel07 this is now ready for review.

File: pkg/operator/metriccontroller/fsync_controller.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	return factory.New().ResyncEvery(1*time.Minute).WithSync(c.sync).ToController("FSyncController", recorder)
+}
+
+func (c *FSyncController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	client, err := getPrometheusClient(ctx, c.secretClient)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
```
Comment (deads2k) [inline pkg/operator/metriccontroller/fsync_controller.go:42]: does returning an error from this method mean the operator goes degrade?  if so, this should not error.

File: pkg/operator/metriccontroller/fsync_controller.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	// First check how many leader changes we see in last 15 minutes
+	etcdLeaderChangesResult, _, err := client.Query(ctx, "increase((max by (job) (etcd_server_leader_changes_seen_total) or 0*absent(etcd_server_leader_changes_seen_total))[5m:1m])", time.Now())
+	if err != nil {
+		return err
```
Comment (deads2k) [inline pkg/operator/metriccontroller/fsync_controller.go:48]: we don't want to degrade if we cannot query.

File: pkg/operator/metriccontroller/fsync_controller.go (hunk 1 above)
Comment (mfojtik) [reply pkg/operator/metriccontroller/fsync_controller.go:42]: it does not in this case.

**Comment by deads2k** [comment]
mfojtik says it doesn't go degraded, just gives an event and logs.  I like events.

I think at some point fsync is worth on its own, but let's start somewhere

/lgtm

**Pull request merged**